.github/workflows/sanitize.yml

```diff
@@ -0,0 +1,132 @@
+name: Sanitize
+
+on:
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+    name: "Sanitize"
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        name:      ["Linux Clang -fsanitize ${{ matrix.sanitizer }}"]
+        os:        ["ubuntu-20.04"]
+        cc:        ["clang-10"]
+        cxx:       ["clang++-10"]
+        py:        ["3.9"]
+        cmake:     ["3.19.x"]
+        mpi:       ["ON"]
+        simd:      ["ON", "OFF"]
+        sanitizer: ["address", "leak", "undefined", "memory", "thread"]
+    env:
+        CC:         ${{ matrix.cc }}
+        CXX:        ${{ matrix.cxx }}
+        # We set PYTHONPATH instead of installing arbor to avoid distribution/OS specific behaviour.
+        PYTHONPATH: ${{ github.workspace }}/build/python
+        # This is a workaround for the unfortunate interaction of MacOS and OpenMPI 4
+        # See https://github.com/open-mpi/ompi/issues/6518
+        OMPI_MCA_btl: "self,tcp"
```

Comment:
I know this has been copied from the other workflow, and perhaps leaving it in is best to avoid future surprises, but we shouldn't need this for the Ubuntu image.

Reply:
Yes, that can go. I do not expect us going for more MacOS.